From: Ben Hutchings <ben@decadent.org.uk>
Subject: driver core: Avoid ABI change for removal of __must_check

Surprisingly, __must_check contributes to the symbol version hash
despite making no real difference to the function's ABI.

--- a/include/linux/device.h
+++ b/include/linux/device.h
@@ -510,8 +510,13 @@
 	struct dev_ext_attribute dev_attr_##_name = \
 		{ __ATTR(_name, _mode, device_show_ulong, device_store_ulong), &(_var) }
 
+#ifdef __GENKSYMS__
+extern int __must_check device_create_file(struct device *device,
+					const struct device_attribute *entry);
+#else
 extern int device_create_file(struct device *device,
 			      const struct device_attribute *entry);
+#endif
 extern void device_remove_file(struct device *dev,
 			       const struct device_attribute *attr);
 extern int __must_check device_create_bin_file(struct device *dev,
